Cannot find simdjson header when library is found

## Description

In `cmake` log can check the simdjson library is found

```
-- SIMDJSON found
```

, but later `ninja` command failed with

```
WasmEdge/test/spec/spectest.cpp:20:10: fatal error: 'simdjson.h' file not found
```

### Current State

<!-- Please shortly describe the bug you have found -->
simdjson.h file not found, build failed

### Expected

<!-- Please shortly describe what you expect instead -->
If library is found, build should pass

## Environment

- Hardware Architecture: aarch64
- Operating system: MacOS 13.4.1

The following information is optional. Please provide them only if you have built from source.
- C++ Compiler version: AppleClang 14.0.3.14030022
- CMake version: 3.26.1
- CMake flags: `cmake -DCMAKE_BUILD_TYPE=Debug -DWASMEDGE_BUILD_TESTS=ON -GNinja ..`
- Path of header: `/usr/local/include/simdjson.h`

## Steps to Reproduce

<!-- Please provide as much information as necessary to reproduce the bug. -->
```shell
mkdir build && cd build
cmake -DCMAKE_BUILD_TYPE=Debug -DWASMEDGE_BUILD_TESTS=ON -GNinja ..
ninja
```

--- a/plugins/wasi_nn/CMakeLists.txt
+++ b/plugins/wasi_nn/CMakeLists.txt
@@ -143,7 +143,7 @@ target_include_directories(wasmedgePluginWasiNN
 
 if(BACKEND STREQUAL "ggml")
   target_include_directories(wasmedgePluginWasiNN PUBLIC ${CMAKE_BINARY_DIR}/_deps/llama-src)
-  target_link_libraries(wasmedgePluginWasiNN PRIVATE common simdjson)
+  target_link_libraries(wasmedgePluginWasiNN PRIVATE common simdjson::simdjson)
   if(WASMEDGE_PLUGIN_WASI_NN_GGML_LLAMA_METAL)
     add_custom_command(
       TARGET wasmedgePluginWasiNN

--- a/test/spec/CMakeLists.txt
+++ b/test/spec/CMakeLists.txt
@@ -86,7 +86,7 @@ wasmedge_add_library(wasmedgeTestSpec
 
 target_link_libraries(wasmedgeTestSpec
   PRIVATE
-  simdjson
+  simdjson::simdjson
   PUBLIC
   std::filesystem
   wasmedgeCommon

